src/types.rs

```diff
@@ -5,16 +5,32 @@ use tracing::{info, warn};
 
 use crate::ScannerError;
 
-#[derive(Debug, Clone)]
+/// Messages streamed by the scanner to subscribers.
+///
+/// Each message represents either data (logs), an error, or a notification about the scanner's
```

Comment:
nit:
```suggestion
/// Each message represents either data or a notification about the scanner's
```
No more error :D 

Also not always logs?